### PR TITLE
Enhancement: Boxes Indexer Integration test

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="ahangsu/indexer-integration-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: once algorand-sdk-testing PR is merged, do
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

File: tests/cucumber/integration.tags [hunk 2]
```diff
@@ -2,6 +2,7 @@
 @algod
 @applications
 @applications.boxes
+@applications.boxes_indexer_confirmed
```
Comment (tzaffi) [inline tests/cucumber/integration.tags:5]: this tag may go away?

**Review by tzaffi — COMMENTED** [review]
Basically approvable but waiting on decision for future of tag `@applications.boxes_indexer_confirmed`

File: tests/cucumber/integration.tags (hunk 2 above)
Comment (ahangsu) [reply tests/cucumber/integration.tags]: gone

**Review by tzaffi — APPROVED** [review]
LGTM

**Comment by tzaffi** [comment]
> LGTM

LEBTM

File: tests/cucumber/steps/steps.js [hunk 3]
```diff
@@ -4439,14 +4443,29 @@ module.exports = function getSteps(options) {
...
+            .getApplicationBoxByName(this.currentApplicationIndex)
+            .name(boxKey)
+            .do();
+        } else if (fromClient === 'indexer') {
+          resp = await this.indexerV2client
+            .lookupApplicationBoxByIDandName(this.currentApplicationIndex)
+            .name(boxKey)
+            .do();
+        } else {
+          assert(
```
Comment (algochoi) [inline tests/cucumber/steps/steps.js:4463]: I think you can `assert.fail` here and in the other two assertions below:

```
assert.fail(
          `expecting algod or indexer, got ${fromClient}`
        );
```

**Review by algochoi — COMMENTED** [review]
Minor comment

File: tests/cucumber/steps/steps.js [hunk 4]
```diff
@@ -4485,6 +4515,69 @@ module.exports = function getSteps(options) {
...
+          .do();
+      } else if (fromClient === 'indexer') {
+        resp = await this.indexerV2client
+          .searchForApplicationBoxes(this.currentApplicationIndex)
+          .limit(limit)
+          .do();
+      } else {
+        assert(
+          false,
+          ''.join(['expecting algod or indexer, got ', fromClient])
```
Comment (algochoi) [inline tests/cucumber/steps/steps.js:4535]: Nice to unify here: ```assert.fail(`expecting algod or indexer, got ${fromClient}`);```

File: tests/cucumber/steps/steps.js [hunk 5]
```diff
@@ -4485,6 +4515,69 @@ module.exports = function getSteps(options) {
...
+          .getApplicationBoxes(this.currentApplicationIndex)
+          .do();
+      } else if (fromClient === 'indexer') {
+        resp = await this.indexerV2client
+          .searchForApplicationBoxes(this.currentApplicationIndex)
+          .do();
+      } else {
+        assert(
+          false,
+          ''.join(['expecting algod or indexer, got ', fromClient])
```
Comment (algochoi) [inline tests/cucumber/steps/steps.js:4560]: ```assert.fail(`expecting algod or indexer, got ${fromClient}`);```

**Review by algochoi — COMMENTED** [review]
LGTM - just a reminder for the other `assert.fail`s

File: tests/cucumber/steps/steps.js (hunk 4 above)
Comment (ahangsu) [reply tests/cucumber/steps/steps.js]: good find, my trash code on js

File: tests/cucumber/steps/steps.js (hunk 5 above)
Comment (ahangsu) [reply tests/cucumber/steps/steps.js]: good find, my trash code on js

**Review by algochoi — APPROVED** [review]

**Pull request merged**